presto-spi/src/main/java/io/prestosql/spi/StandardErrorCode.java

```diff
@@ -64,6 +64,47 @@
     INVALID_SPATIAL_PARTITIONING(0x0000_0029, USER_ERROR),
     INVALID_ANALYZE_PROPERTY(0x0000_002A, USER_ERROR),
     TYPE_NOT_FOUND(0x0000_002B, USER_ERROR),
+    CATALOG_NOT_FOUND(0x0000_002C, USER_ERROR),
```

Comment:
As a pre-commit, we might want to change these to decimals, similar to what we now do for connectors:
```java
GENERIC_USER_ERROR(userCode(0), USER_ERROR),
SYNTAX_ERROR(userCode(1), USER_ERROR),
ABANDONED_QUERY(userCode(2), USER_ERROR),

private static int userCode(int offset)
{
    return offset; // 0x0000_0000
}

private static int internalCode(int offset)
{
    return 0x0001_0000 + offset;
}

private static int resourcesCode(int offset)
{
    return 0x0002_0000 + offset;
}

Reply:
I don't think we should rely on specific "offsets" to indicate the different types of errors. I understand that's how we did it originally, but the error category is more robust. Error codes should be opaque and we shouldn't ascribe any specific meaning to which range an error falls in. It's also easier to search the codebase for the corresponding error when it appears as a literal instead of being derived programmatically. I do agree that we should display them as simple integers instead of hex, though.

Reply:
Maybe “offset” is the wrong name for the parameter. My point is we can replace the hex values with numbers. But we need to preserve the existing mapping and this is an easy way to do it. But a straight conversion from hex to decimal is bad because the higher value ones will have a seemingly random value (which is only understandable in hex).